pkg/client/retry.go

```diff
@@ -42,3 +47,25 @@ func CreateRetryGenerateNameWithFunc(obj kbclient.Object, createFn func() error)
 		return createFn()
 	}
 }
+
+func GetRetriableWithCacheLister(lister cache.GenericNamespaceLister, name string, retriable func(error) bool) (runtime.Object, error){
+	var clusterObj runtime.Object
+	getFunc := func () error {
+		var err error
+		clusterObj, err = lister.Get(name)
+		return err
+	}
+	err := retry.OnError(retry.DefaultRetry, retriable, getFunc)
```

Comment:
I don't think we want to use DefaultRetry:
```
var DefaultRetry = wait.Backoff]{
	Steps:    5,
	Duration: 10 * time.Millisecond,
	Factor:   1.0,
	Jitter:   0.1,
}
```
This retries 5 times with a constant wait time of 10 ms between waits, so if the controller responsible for making the resource ready takes longer than .05 seconds, we'll still fail.

DefaultBackoff would be better, but if we want to make sure we wait a full minute or two, we may want to define our own Backoff struct to ensure that the last attempt happens after at least a minute or so.